### PR TITLE
Add new scim/sync files

### PR DESCRIPTION
**Pull request by Firstyear**
Relates #735 #734 #736 #211 . This adds a lot of the foundactions for a "sync account" which will be used to import and drive the process of actually pushing entries from an external IDM to Kanidm. 

- [ x ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ x ] design document included (if relevant)

### CODE REVIEW

**Review by yaleman — CHANGES_REQUESTED** [review]
Just that minor naming thing I brought up 👍🏻

File: kanidmd/lib/src/identity.rs [hunk 1]
```diff
@@ -133,6 +134,7 @@ pub enum IdentityId {
     // Time stamp of the originating event.
     // The uuid of the originiating user
     User(Uuid),
+    Synch(Uuid),
```
[the remark this answers is not in the document]
Comment (Firstyear) [reply kanidmd/lib/src/identity.rs:137]: Sync is a trait / reserved in rust :)

**Comment by Firstyear** [comment]
I'm also not sure about the name "sync account" as a whole. Like if you were thinking about " ohhh I'm syncing from an external idm into kanidm, what would that be?". Sync Account? Sync Agreement? Sync Connector? Something else? I think it's probably better to change the name sooner than later.

**Comment by yaleman** [comment]
> I'm also not sure about the name "sync account" as a whole. Like if you were thinking about " ohhh I'm syncing from an external idm into kanidm, what would that be?". Sync Account? Sync Agreement? Sync Connector? Something else? I think it's probably better to change the name sooner than later.

Export/Import? Provisioning? Maybe just call it SCIM since it's a catch-all term (though that can cause its own confusion?)

**Comment by Firstyear** [comment]
> > I'm also not sure about the name "sync account" as a whole. Like if you were thinking about " ohhh I'm syncing from an external idm into kanidm, what would that be?". Sync Account? Sync Agreement? Sync Connector? Something else? I think it's probably better to change the name sooner than later.
> 
> Export/Import? Provisioning? Maybe just call it SCIM since it's a catch-all term (though that can cause its own confusion?)

But it's not "scim" as a whole, just for "sync" from an external source which happens to use a single part of the scim interface.

I think for now I'll leave it as is since it's not fully complete as a feature yet and not user facing, so we can rename it later.

**Review by yaleman — APPROVED** [review]

**Pull request merged**